### PR TITLE
etcd-runner: initial commit

### PR DESCRIPTION
**Pull request by xiang90**
/cc @heyitsanthony @gyuho 

We plan to run this against a no consistency checking failure injection cluster. Interesting!

### CODE REVIEW

**Comment by xiang90** [comment]
@heyitsanthony 
```
finished 100, took 30.321140717s
finished 200, took 2.261292425s
finished 300, took 6.206006627s
finished 400, took 2.701600763s
finished 500, took 5.363796536s
finished 600, took 1.926104277s
2016/04/23 06:09:02 no progress after 1 minute!
```

Running this on the failure injection cluster detected the "deadlock" issue.

**Comment by gyuho** [comment]
LGTM. Defer to @heyitsanthony

**Comment by xiang90** [comment]
and this happened 

```
finished 100, took 2.631785858s
2016/04/23 06:30:09 bad lock
```

File: tools/functional-tester/etcd-runner/main.go [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+	var (
+		mu  sync.Mutex
+		cnt int
+	)
+	ctx := context.Background()
+
+	var wg sync.WaitGroup
+
+	for i := range racers {
+		racers[i] = concurrency.NewMutex(randClient(eps), prefix)
```
Comment (heyitsanthony) [inline tools/functional-tester/etcd-runner/main.go]: this leaks a client

Comment (xiang90) [inline tools/functional-tester/etcd-runner/main.go]: i think it is not? We are creating N racers and each racer should have its own client, right?

**Comment by xiang90** [comment]
@heyitsanthony Rebase with master. Can you take a look? Thanks!

File: tools/functional-tester/etcd-runner/main.go (hunk 1 above)
Comment (heyitsanthony) [inline tools/functional-tester/etcd-runner/main.go]: Yes, but `randClient` creates a new client each call and there's no corresponding `Close`.

Comment (xiang90) [inline tools/functional-tester/etcd-runner/main.go]: yea... but this test should be a long running one and should "never" end... so i am less worry about it.

Comment (xiang90) [inline tools/functional-tester/etcd-runner/main.go]: i will get a client array and close it anyway.

Comment (heyitsanthony) [inline tools/functional-tester/etcd-runner/main.go]: If it's long running this is probably the way to go-- the connection leaks would probably eventually break it.

**Comment by heyitsanthony** [comment]
@xiang90 I feel that the lock code belongs in benchmark tool code / this should make better use of / improve the benchmark tool. Is there anything else planned for etcd-runner?

**Comment by xiang90** [comment]
the goal of the lock/lease stuff is not for benchmarking, but for correctness testing against both server-side errors and client side errors. in the future, we might add drop rate where we just kill a client and see how the lock get auto released, etc..

**Comment by xiang90** [comment]
Also for benchmark, we do not expect to run it forever. but for etcd-runner, it should probably just run forever until failed. And then we need to check the log to figure out what happened.

**Comment by heyitsanthony** [comment]
@xiang90 ok lgtm

**Comment by xiang90** [comment]
@heyitsanthony But I will add a lock benchmark to benchmark tool too...

**Pull request merged**